src/main/java/kh/farrukh/springactional/teacher/TeacherServiceImpl.java

```diff
@@ -40,4 +40,17 @@ public Teacher findTeacher(Long id) {
         .orElseThrow(() -> new RuntimeException("teacher.not_found"));
   }
 
+  @Override
+  public void changeBonusAccordingToExamResult(Teacher teacher, Long result) {
+    //if teacher doesn't have bonus, but exam results are good, add bonus
+    if (Boolean.FALSE.equals(teacher.getHasBonus()) && result >= MINIMUM_RESULT_FOR_BONUS) {
+      teacher.setHasBonus(true);
+    }
+
+    //if teacher already has bonus, but exam results are bad, remove bonus
+    if (Boolean.TRUE.equals(teacher.getHasBonus()) && result < MINIMUM_RESULT_FOR_BONUS) {
+      teacher.setHasBonus(false);
```

Comment:
Here you set new value to hasBonus, but didn't call `teacherRepository.save(teacher)`.
So these changes may not be applied to the database

Reply:
But changes are being applied to database on each call of this method.
But I don't know how :smile:.

Reply:
Smells like a new article...